AMS refactor

tofu-rocketry opened the pull request:
This is a pull request back to GC's branch as I'd like to have feedback on my changes. I've attempted to make the code for AMS support a bit simpler and easier for the user.

- Collapse options to switch between STOMP message brokers and AMS to a single config option to simplify things for the user.
- Rename AMS project config option to 'ams_project' to make purpose clear. This also means that it doens't need to go into a completely separate AMS section.
- Remove unnecessary comments from receiver config.
- Update host & port for broker in receiver config (not done for sender to avoid misconfiguration at a site cuasing problems centrally).
- Collapse destination_type into protocol to simplify code. `MQ-BROKER` converts to `STOMP`, and `HTTPS` converts to `AMS`.
- Change name of section that variables are fetched form the config to match new names.
- Remove some unnecessary checks and reorder arguments.
- Remove unncessary `try...except` that raises the exception without modifying it.
- Replace magic strings used for protocols with constants defined in the Ssm2 class as this is good practice.
- Merge `_pull_msg_rest_ams` method into `pull_msg_rest`. Now that the `pull_msg_rest` method has been simplified, the split with the `..._ams` method is a bit unnecessary.
- Add separate token argument to Ssm2 to remove overloading of password
argument as password and username seem redundant and could be removed at
a later stage.

tofu-rocketry committed:
Simplify AMS options and tidy up config files

- Collapse options to switch between STOMP message brokers and AMS to a
single config option to simplify things for the user.
- Rename AMS project config option to 'ams_project' to make purpose
clear. This also means that it doens't need to go into a completely
separate AMS section.
- Remove unnecessary comments from receiver config.
- Update host & port for broker in receiver config (not done for sender
to avoid misconfiguration at a site cuasing problems centrally).

tofu-rocketry committed:
Collapse destination_type into protocol

- Collapse destination_type into protocol to simplify code. MQ-BROKER
converts to STOMP, and HTTPS converts to AMS.
- Change name of section that variables are fetched form the config to
match new names.
- Remove some unnecessary checks and reorder arguments.

tofu-rocketry committed:
Remove unncessary try...except

Remove unncessary try...except that raises the exception without
modifying it.

tofu-rocketry committed:
Replace magic protocol strings with constants

Replace magic strings used for protocols with constants defined in the
Ssm2 class as this is good practice.

tofu-rocketry committed:
Merge _pull_msg_rest_ams method into pull_msg_rest

Now that the pull_msg_rest method has been simplified, the split with
the ..._ams method is a bit unnecessary.

tofu-rocketry committed:
Add separate token argument to Ssm2

Add separate token argument to Ssm2 to remove overloading of password
argument as password and username seem redundant and could be removed at
a later stage.

tofu-rocketry committed:
Add AMS exception to errors caught

Rather than aliasing the AMS exception to the STOMP one, which would
possibly break things if for some reason you have AMS installed but
weren't using it with SSM, add it to the list of exceptions caught.

tofu-rocketry committed:
Remove unused AMS import

tofu-rocketry committed:
Remove setting of AMS and STOMP defaults manually

The variables are either required in the config and are protected by a
try...except when fetching the config, or are better set nearer to where
they're needed.

tofu-rocketry committed:
Rename pull_msg_rest to pull_msg_ams

Rename pull_msg_rest to pull_msg_ams to match other refactoring

gregcorbett commented on bin/receiver.py:
```diff
@@ -104,36 +104,26 @@ def main():
     log.info(LOG_BREAK)
     log.info('Starting receiving SSM version %s.%s.%s.', *__version__)
 
     # Determine the protocol and destination type of the SSM to configure.
```
as there is no longer a destination type, this comment should be reworded

tofu-rocketry replied:
Done.

gregcorbett commented:
> Update host & port for broker in receiver config (not done for sender to avoid misconfiguration at a site cuasing problems centrally).

It looks like you've updated host and port in the sender config and not in the receiver config?

More generally, do we want to "specify" a host and port in either config at all? It changes between sending via STOMP or via AMS, so might get confusing. Or at least a comment saying the details are for a specific means of sending as per the bdii and network settings?

gregcorbett commented:
The AMS documentation in README.md and migrating_to_ams_broker.md should be updated to reflect the changes in the config files.

tofu-rocketry committed:
Improve comments and remove host & port examples

- Remove referece to destination type in comment about getting protocol
as the former is no longer used.
- Clarify that host and port are an alterative to bdii and network
unless using STOMP in which case it is mandatory.
- Remove example hosts and ports as they're different depending on if
using AMS or STOMP.
- Change network in receiver.cfg to PROD as TEST-NWOB is no more.

tofu-rocketry committed:
Update docs to match refactoring

Update docs to match refactoring and fix a few small issues.

tofu-rocketry commented:
Fair points. I've addressed them all.

tofu-rocketry committed:
Rename migrating to ams file and update refs

Rename migrating_to_ams_broker.md documentation file to remove _broker
as we are informed that AMS is not a broker. Update references to this
file.
